var/spack/repos/builtin/packages/nlcglib/package.py

```diff
@@ -71,6 +83,29 @@ def cmake_args(self):
             options += [self.define("LAPACK_VENDOR", "MKL")]
         elif self.spec["blas"].name in ["intel-oneapi-mkl"]:
             options += [self.define("LAPACK_VENDOR", "MKLONEAPI")]
+            mkl_mapper = {
+                "threading": {"none": "sequential", "openmp": "gnu_thread", "tbb": "tbb_thread"},
+                "mpi": {"intel-mpi": "intelmpi", "mpich": "mpich", "openmpi": "openmpi"},
+            }
+
+            mkl_threads = mkl_mapper["threading"][
+                self.spec["intel-oneapi-mkl"].variants["threads"].value
+            ]
+
+            mpi_provider = self.spec["mpi"].name
+            if mpi_provider in ["mpich", "cray-mpich", "mvapich", "mvapich2"]:
+                mkl_mpi = mkl_mapper["mpi"]["mpich"]
+            else:
+                mkl_mpi = mkl_mapper["mpi"][mpi_provider]
+
+            options.extend(
+                [
+                    self.define("MKL_INTERFACE", "lp64"),
+                    self.define("MKL_THREADING", mkl_threads),
+                    self.define("MKL_MPI", mkl_mpi),
+                ]
+            )
+
```

Comment:
I just want to note that you may be better off long term moving to using spack's generic blas/lapack flags because it would give you support for any blas/lapack provided by spack without having to add explicit support in the package itself. We've done this e.g. in DLA-Future with the latest versions: https://github.com/spack/spack/blob/5e8eff24d2b6b13cf576a84d746795c6a0b6c610/var/spack/repos/builtin/packages/dla-future/package.py#L221-L228.

Also not a blocker, especially if this is currently not possible to use because of how nlcglib's CMake works.

Reply:
I'll postpone this. CMake needs to be changed at nlcglib and the release relying on mkl config is already out.